sunpy/coordinates/screens.py

```diff
@@ -39,6 +42,73 @@ def __exit__(self, exc_type, exc_value, exc_tb):
             raise RuntimeError(f"Cannot remove {self._context_name} from tracking stack because {removed} is last active.")
         Helioprojective._assumed_screen = self._old_assumed_screen
 
+    def _iterate_calculate_distance(self, coord, distance, screen_frame):
+        """
+        Numerically calculates the distance component to a screen to promote a
+        coordinate from 2D to 3D.
+
+        This method should be used only if the screen has been differentially rotated,
+        since otherwise promoting a coordinate from 2D to 3D is straightforward.
+
+        The approach used for numerical solving is to iterate to the solution starting
+        with an initial guess. For each guess, we construct a 3D coordinate and perform
+        a 3D->2D->3D transformation in the native frame of the screen. The amount that
+        the coordinate shifts in the 3D->2D->3D transformation is related to how far off
+        the guess is: when the shift is zero, the guess is correct. To form the next
+        guess, we linearly interpolate/extrapolate between the latest guess and the
+        closest guess over past iterations to estimate where the shift will be zero.
+
+        The maximum number of iterations is hard-coded to 20, but convergence will
+        typically take no more than 6 iterations. When working with an array coordinate,
+        elements that have converged are skipped in subsequent iterations.
+        """
+        # Check if the logging level is at least DEBUG (for performance reasons)
+        debug_output = log.getEffectiveLevel() <= logging.DEBUG
+
+        log.debug("Differentially rotating the screen")
+        use = np.ones(distance.shape, dtype=bool)
+        delta = np.empty_like(distance)
+        next_distance = np.empty_like(distance)
+        for niter in range(20):
+            # If this is not the first iteration, update best distance and calculate new distance
+            if niter > 0:
+                # Calculate for only those pixels that do not meet the tolerance
+                with np.errstate(invalid='ignore'):
+                    use = np.abs(delta / distance) >= 1e-11*u.one
```

Comment:
Is there any particular rationale for using 1e-11?

Reply:
Heh, yes, two reasons:

- It's roughly 1 meter at 1 AU (actually, 1.5 meters)
- `delta` is the subtraction of two `distance`-sized values, so I wouldn't want to go below 1e-12 due to numerical precision

Reply:
Added